client/lib/wp/offline-library/index.js

```diff
@@ -0,0 +1,102 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import stringify from 'json-stable-stringify';
+
+const readCache = () => {
+	try {
+		return require( '../../../../cached-requests.json' );
+	} catch ( e ) {
+		return {};
+	}
+};
+
+const saveRequests = requests => {
+	const requestData = {};
+
+	requests.forEach( ( value, key ) => {
+		const path = JSON.parse( key ).path;
+
+		if ( ! requestData.hasOwnProperty( path ) ) {
+			requestData[ path ] = {};
+		}
+
+		requestData[ path ][ key ] = value;
+	} );
+
+	const file = new Blob( [ JSON.stringify( requestData, null, 2 ) ], {
+		type: 'application/json',
+	} );
+
+	const url = URL.createObjectURL( file );
+	window.open( url );
+};
+
+export const makeOffline = wpcom => {
+	const location = window.location;
+	const primingRequested = /[&?]wpcom_priming=1/.test( location );
+	const offlineRequested = /[&?]wpcom_offline=1/.test( location );
+
+	if ( ! primingRequested && ! offlineRequested ) {
+		return wpcom;
+	}
+
+	// eslint-disable-next-line no-console
+	primingRequested && console.log( 'Priming wpcom request cache' );
+
+	offlineRequested &&
+		// eslint-disable-next-line no-console
+		console.log(
+			'Delivering wpcom requests from cache\n' +
+				'Run `saveRequests()` in the developer console to save cache file.'
+		);
+
+	const request = wpcom.request.bind( wpcom );
+	const requests = new Map();
+
+	const storedRequests = readCache();
+	Object.keys( storedRequests ).forEach( path => {
+		Object.keys( storedRequests[ path ] ).forEach( key => {
+			requests.set( key, storedRequests[ path ][ key ] );
+		} );
+	} );
+
+	window.saveRequests = () => saveRequests( requests );
+
+	Object.defineProperty( wpcom, 'request', {
+		value: ( params, callback ) => {
+			if ( offlineRequested ) {
+				const stored = requests.get( stringify( params ) );
+
+				if ( undefined === callback ) {
+					if ( undefined === stored ) {
+						return Promise.reject( new Error( 'Network inaccessible' ) );
```

Comment:
Here and below, this error message could include something about offline requests.  "Network inaccessible and request for [path] not cached."

Reply:
good point - do you think this would be a good warrant for a console message as well? maybe a developer doesn't realize it wasn't primed?

Reply:
Sure, this would also keep the error message concise, while providing all the relevant information somewhere.  +1 for a console message like "Request [URL] was not found in offline requests cache."

Reply:
in 9107268 I added a console message and changed the error text